OPHD/States/CrimeExecution.h

```diff
@@ -10,10 +11,13 @@ class CrimeExecution
 public:
 	CrimeExecution(NotificationArea& notificationArea);
 
+	void difficulty(Difficulty difficulty) { mDifficulty = difficulty; }
+
 	void executeCrimes(const std::vector<Structure*>& structuresCommittingCrime);
 
 	void executeStealingFood(FoodProduction& structure);
 
 private:
+	Difficulty mDifficulty{ Difficulty::Medium };
```

Comment:
I feel like this should probably be a constructor parameter.

Reply:
Actually, make it a constructor parameter, and declare it as `const`, so we know if it was ever not set, as unset `const` values will produce a compiler warning.

I don't think it makes sense to change that during a game.

Reply:
This was my initial design. C++ makes it difficult though. `MapViewState` is constructed before saved games are loaded. So, to pass `CrimeExecution` the difficulty setting in the constructor initialization list of `MapViewState`, we would probably need to load the saved game before constructing `MapViewState`. Or make `CrimeExecution` a `unique_ptr` so it can be initialized after MapViewState is constructed. 

It is not uncommon to allow a player to change difficulty settings mid play-through in some games, wasn't sure what Leeor's plan was here.

Reply:
Oh, I don't think difficulty is actually set in game yet, so sort of academic in some respects for me to be adding it to crime. I wasn't ready to add a GUI setting for difficulty on the menu if someone else wanted to tackle that.

Reply:
Actually, you bring up a good point. I think the way new games versus loaded games are being initialized needs to be refactored. There's a lot of modification of effectively static values after construction due to the way the current code is structured. It's one of those things I've wanted to address, but never had the time. I suspect it will be quite a big ball to unravel.

Anyway, I suppose we have to stick with something that works with the current design.